File: src/main/java/dev/tang/tool/logger/Logger.java
package dev.tang.tool.logger;

import dev.tang.tool.util.FileUtils;
import dev.tang.tool.util.PathUtils;
import dev.tang.tool.util.TimeUtils;

/**
 * 日志显示器
 */
public class Logger {

    private Class<?> clazz;

    public Logger(Class<?> clazz) {
        this.clazz = clazz;
    }

    public void warn(String content) {
        String loggerPath = PathUtils.getLoggerPath();
        content = "[Warn] " + TimeUtils.getCurrentTime() + " [" + clazz.getName() + "] " + content;
        FileUtils.writeFile(loggerPath, content, true);
    }

    public void warn(String content, Object... args) {
        for (int i = 0; i < args.length; i++) {
            content = content.replace("{" + i + "}", String.valueOf(args[i]));
        }

        String loggerPath = PathUtils.getLoggerPath();
        content = "[Warn] " + TimeUtils.getCurrentTime() + " [" + clazz.getName() + "] " + content;
        FileUtils.writeFile(loggerPath, content, true);
    }

    public void error(String content, Object... args) {
        for (int i = 0; i < args.length; i++) {
            content = content.replace("{" + i + "}", String.valueOf(args[i]));
        }

        String loggerPath = PathUtils.getLoggerPath();
        content = "[Error] " + TimeUtils.getCurrentTime() + " [" + clazz.getName() + "] " + content;
        FileUtils.writeFile(loggerPath, content, true);
    }
}

File: src/main/java/dev/tang/tool/logger/Logger.kt
package dev.tang.tool.logger

import dev.tang.tool.util.FileUtils
import dev.tang.tool.util.PathUtils
import dev.tang.tool.util.TimeUtils

/**
 * 日志显示器
 */
class Logger(private val clazz: Class<*>) {

    fun warn(content: String) {
        var content = content
        val loggerPath = PathUtils.loggerPath
        content = "[Warn] " + TimeUtils.currentTime + " [" + clazz.name + "] " + content
        FileUtils.writeFile(loggerPath, content, true)
    }

    fun warn(content: String, vararg args: Any) {
        var content = content
        for (i in args.indices) {
            content = content.replace("{$i}", args[i].toString())
        }

        val loggerPath = PathUtils.loggerPath
        content = "[Warn] " + TimeUtils.currentTime + " [" + clazz.name + "] " + content
        FileUtils.writeFile(loggerPath, content, true)
    }

    fun error(content: String, vararg args: Any) {
        var content = content
        for (i in args.indices) {
            content = content.replace("{$i}", args[i].toString())
        }

        val loggerPath = PathUtils.loggerPath
        content = "[Error] " + TimeUtils.currentTime + " [" + clazz.name + "] " + content
        FileUtils.writeFile(loggerPath, content, true)
    }
}
